### PR TITLE
Writing release notes and bump version to 0.36.0

### PR DESCRIPTION
**Pull request by SimonBiggs**
Here are the release notes prepared for the `0.36.0` release. The full change log between current main and version `0.35.0` can be seen over at https://github.com/pymedphys/pymedphys/pull/1443.

Keen for all those who were involved in the changes in this release to have a look over and give feedback. I'm aiming to make the release on Friday next week, but there's nothing specific about that date if it does need to be delayed.

Pinging @pchlap @sjswerdloff @dg1an3 @Matthew-Jennings and @rembishj. If there's anything that you've changed that is `pymedphys` user facing that I've missed please feel free to edit this PR yourselves directly. If there's any rewording or clarifying or fixing, also feel free to jump in. Also, if there's anything newsworthy feel free to have it announced in `## News around this release`. At some point @pchlap if you're interested you can give a shout out to https://github.com/pyplati/platipy within that section if you think that's appropriate.

### CODE REVIEW

**Review by pchlap — APPROVED** [review]
LGTM

@SimonBiggs would definitely be keen to shout out platipy at some point, but that needs a little bit more work first :)

**Comment by SimonBiggs** [comment]
@pchlap once https://github.com/pymedphys/pymedphys/pull/1431 lands, would you be able to add a line in this PR to account for it?

File: lib/pymedphys/docs/release-notes.md [hunk 1]
```diff
@@ -46,7 +46,8 @@ This project adheres to
 ### New features and enhancements
 
 * Added CLI argument for setting the hostname on the DICOM listen server. For
-  example `pymedphys dicom listen --host 127.0.0.1`.
+  example `pymedphys dicom listen --host 127.0.0.1 7779`.
```
Comment (SimonBiggs) [inline lib/pymedphys/docs/release-notes.md:49]: Does the `7779` need to be proceeded with `--port`?

Comment (pchlap) [reply lib/pymedphys/docs/release-notes.md:49]: No, since thats not an optional parameter.. The user must specify a port...

Comment (SimonBiggs) [reply lib/pymedphys/docs/release-notes.md:49]: Yup, fair enough :slightly_smiling_face: 

Could we change the text to the following:

```suggestion
  example `pymedphys dicom listen 7779 --host 127.0.0.1 `.
```

Optional parameters at the end tend to make more sense to me.

Comment (pchlap) [reply lib/pymedphys/docs/release-notes.md]: Yep, that is nicer.

**Pull request merged**